helix-term/src/ui/editor.rs

```diff
@@ -651,6 +655,43 @@ impl EditorView {
         );
     }
 
+    /// Apply the highlighting on the lines where a cursor is active
+    pub fn highlight_cursorline(doc: &Document, view: &View, surface: &mut Surface, theme: &Theme) {
+        let text = doc.text().slice(..);
+        let last_line = view.last_line(doc);
+
+        let primary_line = doc.selection(view.id).primary().cursor_line(text);
+
+        // The secondary_lines do contain the primary_line, it doesn't matter
+        // as the else-if clause in the loop later won't test for the
+        // secondary_lines if primary_line == line.
+        // It's used inside a loop so the collect isn't needless:
+        // https://github.com/rust-lang/rust-clippy/issues/6164
+        #[allow(clippy::needless_collect)]
+        let secondary_lines: Vec<_> = doc
+            .selection(view.id)
+            .iter()
+            .map(|range| range.cursor_line(text))
+            .collect();
+
+        let primary_style = theme.get("ui.cursorline.primary");
+        let secondary_style = theme.get("ui.cursorline.secondary");
+
+        for line in view.offset.row..(last_line + 1) {
+            let area = Rect::new(
+                view.area.x,
+                view.area.y + (line - view.offset.row) as u16,
+                view.area.width,
+                1,
+            );
+            if primary_line == line {
+                surface.set_style(area, primary_style);
+            } else if secondary_lines.contains(&line) {
+                surface.set_style(area, secondary_style);
+            }
```

Comment:
Why not just highlight `primary_line` after `secondary_line` after the loop so we don't have to check whether or not it was part of secondary line?

Reply:
We still have to check for each `line` if `secondary_lines` contains it, since `line` is one of all the lines on the screen, only some will be highlighted.

Strictly speaking, not checking and then just overwriting after the loop might be even (veery slightly) slower, since the comparison `primary_line == line` is basically free, but coloring the `secondary_line` (which is redundant because it will be overwritten) is less cheap. (I might be wrong though.)